src/ripple/rpc/tests/AccountOffers.test.cpp

```diff
@@ -0,0 +1,275 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2016 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/protocol/JsonFields.h>     // jss:: definitions
+#include <ripple/protocol/Feature.h>
+#include <ripple/test/jtx.h>
+
+namespace ripple {
+namespace test {
+
+class AccountOffers_test : public beast::unit_test::suite
+{
+public:
+    static
+    std::unique_ptr<Config>
+    makeConfig(bool setup_admin)
+    {
+        auto p = std::make_unique<Config>();
```

Comment:
Food for thought: what if we had the `Config` constructor accept a lambda that it invoked inside the constructor body and allows modification of the object before its ever returned?

Something like:

```
return std::make_unique<Config>([setup_admin](Config* cfg)
{
    setupConfigForUnitTests(cfg);
    
    if (! setup_admin)
    {
        (*p)["port_rpc"].set("admin", "");
        (*p)["port_ws"].set("admin", "");
    }
}
```

With only marginal changes and no additional complexity (a `UnitTestConfig` derived class that calls `setupConfigForUnitTests` in its constructor) this common idiom could be dramatically simplified.

No need for changes; as I said, just food for thought.